### PR TITLE
Additional `CompletedRequestMap` constructors and deprecations

### PR DESCRIPTION
**Pull request by kyri-petrou**
This PR aims to improve UX by adding a few more ways to construct a `CompletedRequestMap`

### Deprecation of  `++`, `insert` and `insertOption` methods

The rationale behind this the deprecation of these methods is that prior to version 0.6.1, the only way to populate a `CompletedRequestMap` was by folding over an empty one and immutably inserting all the requests / responses to it. One of the major recent optimizations was for `CompletedRequestMap` to wrap a mutable map, however, in order for mutability not to leak to the public API the `++`, `insert` and `insertOption` methods still use immutable maps. This means that existing user code will not benefit from these optimizations until they use the new constructors added since then.

TLDR; The deprecation of these methods is simply meant as a way to signal to users (who don't read release notes) to avoid usage of these methods as they'll lead to suboptimal performance

### CODE REVIEW

File: zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala [hunk 1]
```diff
@@ -73,29 +100,58 @@ sealed abstract class CompletedRequestMap { self =>
...
   val empty: CompletedRequestMap =
     new Immutable(immutable.HashMap.empty)
 
+  /**
+   * Combines all completed request maps into a single one.
+   *
+   * This method is left-associated, meaning that if a request is present in
+   * multiple maps, it will be overriden by the last map in the list.
+   */
+  def fromCompletedRequestMaps(maps: Iterable[CompletedRequestMap]): CompletedRequestMap = {
```
Comment (kyri-petrou) [inline zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala]: Should I rename this to `combine` perhaps?

File: zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala [hunk 2]
```diff
@@ -133,14 +189,34 @@ object CompletedRequestMap {
     map
   }
 
-  private[query] object unsafe {
+  /**
+   * Constructs a completed requests map containing a single entry
+   */
+  def single[E, A](request: Request[E, A], exit: Exit[E, A]): CompletedRequestMap =
```
Comment (ghostdogpr) [inline zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala:201]: How about an `apply` constructor with one or more values?

File: zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala [hunk 3]
```diff
@@ -133,14 +189,34 @@ object CompletedRequestMap {
...
+    new Immutable(immutable.HashMap((request, exit)))
 
-    def fromSuccesses[E, A, B](requests: Chunk[Request[E, B]], responses: Chunk[B]): CompletedRequestMap =
-      fromWith(requests, responses)(identity, Exit.succeed)
+  /**
+   * Unsafe API for constructing completed request maps.
+   *
+   * Constructing a [[CompletedRequestMap]] via these megthods can improve
```
Comment (ghostdogpr) [inline zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala]:
```suggestion
   * Constructing a [[CompletedRequestMap]] via these methods can improve
```

File: zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala (hunk 1 above)
Comment (ghostdogpr) [reply zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala]: yeah

File: zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala (hunk 2 above)
Comment (kyri-petrou) [reply zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala:201]: Is this what you mean?

```scala
def apply[E, A](results: (Request[E, A], Exit[E, A])*): CompletedRequestMap
```

Comment (ghostdogpr) [reply zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala:201]: Yep

**Review by ghostdogpr — APPROVED** [review]
